### PR TITLE
Leland scala rules

### PR DESCRIPTION
**Pull request by enncoded**
Not ready yet. Just making a PR to check GH action results

### CODE REVIEW

**Comment by minusworld** [comment]
So, I'm realizing after looking at this: we had talked about just adding Scala and Kotlin to the language fields for these Java rules, but we won't be able to run tests on them unless they're in the same folder. We'll either have to add the test cases to the Java folder, or create duplicates (or symlinks) of the rules in the appropriate folder.

File: scala/lang/security/audit/insecure-random.yaml [hunk 1]
```diff
@@ -12,12 +12,16 @@ rules:
       resources:
         - https://find-sec-bugs.github.io/bugs.htm
     message: >-
-      Flags the use of a predictable random value from `scala.util.Random`. This can lead to vulnerabilities when used in security
-      contexts, such as in a CSRF token, password reset token, or any other secret value. To fix this, use java.security.SecureRandom
-      instead.
+      Flags the use of a predictable random value from an insecure random number generator. This can lead to vulnerabilities when used
```
Comment (kurt-r2c) [inline scala/lang/security/audit/insecure-random.yaml:15]: Is this a rule port?  There are non-critical use cases for deterministic random number generators (caching, etc) where `java.util.Random` is acceptable.

File: java/lang/security/audit/crypto/gcm-detection.yaml [hunk 2]
```diff
@@ -7,21 +7,25 @@ rules:
...
       the ciphertext. Reusing same Initialization Vector with the same Key to
       encrypt multiple plaintext blocks allows an attacker to compare the
-      ciphertexts and then, with some  assumptions on the content of the
+      ciphertexts and then, with some assumptions on the content of the
       messages, to gain important information about the data being encrypted.
     patterns:
       - pattern-either:
-          - pattern: $METHOD.getInstance("AES/GCM/NoPadding",...);
-          - pattern: new GCMParameterSpec(...);
+          - pattern: new GCMParameterSpec(...)
```
Comment (kurt-r2c) [inline java/lang/security/audit/crypto/gcm-detection.yaml:29]: This is failing tests:
```
✖ gcm-detection  expected lines: [37, 42, 52, 56], reported lines: [37, 52] 
 test file path: /src/java/lang/security/audit/crypto/gcm-detection.java
```

**Review by kurt-r2c — CHANGES_REQUESTED** [review]
Failing tests for `gcm-detection`
